gcapi/gcapi.py

```diff
@@ -147,11 +149,24 @@ def iterate_all(self, params=None):
             yield from current_list
             offset += req_count
 
-    def detail(self, pk):
-        result = self._client(
-            method="GET", path=urljoin(self.base_path, pk + "/")
-        )
-        self._verify_against_schema(result)
+    def detail(self, pk=None, **params):
+        if len([x for x in [pk, params] if x]) != 1:
+            raise ValueError("Only one of pk or params must be specified")
+
+        if pk is not None:
+            result = self._client(
+                method="GET", path=urljoin(self.base_path, pk + "/")
+            )
+            self._verify_against_schema(result)
```

Comment:
Is there a reason that this verification is done only for "by pk" requests?

Reply:
It's being done in the `page` method https://github.com/DIAGNijmegen/rse-gcapi/pull/86/files#diff-0b0fdcdf902a8e6c2a49092303d186e4ef201b468ff23b4e0ca1446218c6b9b7R128-R137